File: pytools/examples/php/server/protocol/ReqTestXX.php
<?php
namespace protocol;

use network\Packet;


class ReqTestXX
{
	private $idU8 = 0;
	private $idU16 = 0;
	private $idU32 = 0;
	private $repeatIdU8 = [];
	private $optionalIdU8Flag = 0;
	private $optionalIdU8 = 0;


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->idU8 = $packet->readU8();
		$this->idU16 = $packet->readU16();
		$this->idU32 = $packet->readU32();
		$repeatIdU8Count = $packet->readU16();
		for ($i = 0; $i < $repeatIdU8Count; $i++)
		{
			array_push($this->repeatIdU8, $packet->readU8());
		}
		$this->optionalIdU8Flag = $packet->readU8();
		if ($this->optionalIdU8Flag == 1)
		{
			$this->optionalIdU8 = $packet->readU8();
		}
	}


	public function getIdU8()
	{
		return $this->idU8;
	}

	public function getIdU16()
	{
		return $this->idU16;
	}

	public function getIdU32()
	{
		return $this->idU32;
	}

	public function getRepeatIdU8()
	{
		return $this->repeatIdU8;
	}

	public function getOptionalIdU8()
	{
		return $this->optionalIdU8;
	}

}

File: pytools/examples/golang/src/server/proto/req_test_x_x.go
package proto

import (
	"packet"
)

type ReqTestXX struct {
	idU8                     uint8
	idU16                    uint16
	idU32                    uint32
	repeatIdU8               []uint8
	optionalIdU8Flag         uint8
	optionalIdU8             uint8
}

func ReqTestXXDecode(pack *packet.Packet) *ReqTestXX {
	reqTestXX := &ReqTestXX{}

	reqTestXX.idU8 = pack.ReadUint8()
	reqTestXX.idU16 = pack.ReadUint16()
	reqTestXX.idU32 = pack.ReadUint32()
	repeatIdU8Count := pack.ReadUint16()
	for ;repeatIdU8Count > 0; repeatIdU8Count-- {
		reqTestXX.repeatIdU8 = append(reqTestXX.repeatIdU8, pack.ReadUint8())
	}
	reqTestXX.optionalIdU8Flag = pack.ReadUint8()
	if reqTestXX.optionalIdU8Flag == 1 {
		reqTestXX.optionalIdU8 = pack.ReadUint8()
	}
	return reqTestXX
}

func (this *ReqTestXX) GetIdU8() uint8 {
	return this.idU8
}

func (this *ReqTestXX) GetIdU16() uint16 {
	return this.idU16
}

func (this *ReqTestXX) GetIdU32() uint32 {
	return this.idU32
}

func (this *ReqTestXX) GetRepeatIdU8() []uint8 {
	return this.repeatIdU8
}

func (this *ReqTestXX) GetOptionalIdU8() uint8 {
	return this.optionalIdU8
}
